mlperf_logging/package_checker/package_checker.py

```diff
@@ -74,20 +92,32 @@ def _print_divider_bar():
     print('------------------------------')
 
 
-def check_training_result_files(folder, ruleset, quiet, werror, rcp_bypass, rcp_bert_train_samples):
+def check_training_result_files(folder, usage, ruleset, quiet, werror,
+                                rcp_bypass, rcp_bert_train_samples):
     """Checks all result files for compliance.
 
     Args:
         folder: The folder for a submission package.
         ruleset: The ruleset such as 0.6.0, 0.7.0, or 1.0.0
     """
-
     if ruleset == '0.6.0':
-       allowed_benchmarks = _ALLOWED_BENCHMARKS_V06
+       assert(usage == "training")
+       allowed_benchmarks = _TRAINING_BENCHMARKS_V06
     elif ruleset == '0.7.0':
-       allowed_benchmarks = _ALLOWED_BENCHMARKS_V07
+       if usage == "training":
+           allowed_benchmarks = _TRAINING_BENCHMARKS_V07
+       elif usage == "hpc":
+           allowed_benchmarks = _HPC_BENCHMARKS_V07
     elif ruleset == '1.0.0':
-       allowed_benchmarks = _ALLOWED_BENCHMARKS_V10
+       if usage == "training":
+           allowed_benchmarks = _TRAINING_BENCHMARKS_V10
+       elif usage == "hpc":
+           allowed_benchmarks = _HPC_BENCHMARKS_V10
```

Comment:
I have changed this in the spirit you mentioned. I did the same in result summarizer and moved all the benchmark file counts and allowed benchmark dictionaries into a file called benchmark_meta.py. This is used by result summarizer and package checker, so that once more benchmarks are added, just a single file needs to be changed. I hope you like that refactor.

Reply:
I like your refactor